common/tests/test_get_token.py

```diff
@@ -0,0 +1,32 @@
+import unittest
+import os
+
+from common.api import Api
+from common.file_helpers import mkdirs_exists_ok
+from common.basedir import PERSIST
+
+class TestApiGetToken(unittest.TestCase):
+    def is_private_key_generated(self):
```

Comment:
don't copy this code, just abstract it and use it in the tests

Reply:
Should I just check if the private key is generated by checking if id_rsa exists, and not actually try to run commands to generate it?

Reply:
yes, don't generate if already present. The logic for this should all live in registration.py though

Reply:
Okay, Something like this in check private key function :

`assertEqual(os.path.exists(path_to_id_rsa), True)`

And `path_to_id_rsa` will be the path extracted from registration.py

Or use If-else instead of assert 

What to do if private key is not present? Raise an exception and stop?

Reply:
Let me just write the test, that's probably easier. I'll merge the changes to the api